ports/geckolib/glue.rs

```diff
@@ -3620,10 +3619,11 @@ pub extern "C" fn Servo_ReparentStyle(
     let element = element.map(GeckoElement);
 
     let mut cascade_flags = CascadeFlags::empty();
-    if style_to_reparent.is_anon_box() {
-        cascade_flags.insert(CascadeFlags::SKIP_ROOT_AND_ITEM_BASED_DISPLAY_FIXUP);
-    }
     if let Some(element) = element {
+        // NOTE(emilio): This relies on element.is_none() == pseudo.is_none(),
+        // which the caller guarantees, fortunately. But this doesn't handle the
```

Comment:
It is not clear to me how is it guaranteed... Maybe you mean when `element.is_none()` is true, then `pseudo.is_none()` is also true? I'm not sure why we cannot reparent non-pseudo element here.

It is also unclear to me why this relies on that... What would happen if it is not that case?

Reply:
Well, we'd set the flags wrong (we'd get a pseudo with an IS_VISITED flag, potentially, which is not great).

It's guaranteed because of this line:

  https://searchfox.org/mozilla-central/rev/fb5422ff98ba43f3732debd9d1f4dcd3b3a920f6/layout/base/ServoRestyleManager.cpp#1606

Reply:
This seems like it should be `element.is_none() != pseudo.is_none()` then?

Reply:
Well, you're right about what's guaranteed. (when `element.is_some()` in your comment). Will adjust the comment.

Reply:
I adjusted the comment to be correct.

Reply:
(The real assumption is `element.is_some() implies pseudo.is_none()`, for this code to be correct)